lib/shopify_api/api_version.rb

```diff
@@ -1,113 +1,111 @@
 # frozen_string_literal: true
 module ShopifyAPI
-  class ApiVersion
+  class ApiVersion < ActiveResource::Base
     class UnknownVersion < StandardError; end
     class InvalidVersion < StandardError; end
-
-    extend DefinedVersions
-
     include Comparable
 
-    def self.coerce_to_version(version_or_name)
-      return version_or_name if version_or_name.is_a?(ApiVersion)
-
-      @versions ||= {}
-      @versions.fetch(version_or_name.to_s) do
-        raise UnknownVersion, "#{version_or_name} is not in the defined version set: #{@versions.keys.join(', ')}"
+    self.site = "https://app.shopify.com/services/"
+    self.primary_key = :handle
+
+    HANDLE_FORMAT = /^\d{4}-\d{2}$/.freeze
+    UNSTABLE_HANDLE = 'unstable'
+    UNSTABLE_AS_DATE = Time.utc(3000, 1, 1)
+    API_PREFIX = '/admin/api/'
+
+    class << self
+      attr_reader :versions
+
+      def coerce_to_version(version_or_handle)
+        return version_or_handle if version_or_handle.is_a?(ApiVersion)
+        if @versions.nil?
+          warn "[API VERSION WARNING] Known API Version set is empty. Initializing unvalidated version from handle."
+          return ApiVersion.new(handle: version_or_handle)
+        end
+
+        @versions[version_or_handle.to_s].tap do |api_version|
+          unless api_version
+            raise ApiVersion::UnknownVersion, "UnknownVersion API version specified, `#{version_or_handle}`. \n Versions available: #{@versions.keys}"
+          end
+        end
       end
-    end
 
-    def self.define_version(version)
-      @versions ||= {}
+      def define_known_versions
+        @versions = Meta.admin_versions.map { |version| [version.handle, version] }.to_h
+      rescue ActiveResource::ConnectionError => e
+        warn "[API VERSION WARNING] Could not fetch Admin API versions."
+        warn "[API VERSION WARNING] #{e.message}"
+        @versions = nil
+      end
 
-      @versions[version.name] = version
-    end
+      def clear_defined_versions
+        @versions = nil
+      end
 
-    def self.clear_defined_versions
-      @versions = {}
+      def latest_stable_version
+        warn(
+          '[DEPRECATED] ShopifyAPI::ApiVersion.latest_stable_version is deprecated and will be removed in a future version.'
+        )
+        @versions.values.find(&:latest_supported?)
+      end
     end
 
-    def self.latest_stable_version
-      @versions.values.select(&:stable?).sort.last
+    def initialize(attributes = {}, persisted = false)
+      super
+      @attributes['handle'] = @attributes['handle'].to_s
```

Comment:
could handle default to unstable?

Reply:
I'm not sure we want to be in a situation where the default is unstable. unstable should never be used in production and I cant think of a reason you'd want to call `ApiVersion.new`  vs `ApiVersion.new(handle: :unstable)` if you wanted unstable